lib/src/source_visitor.dart

```diff
@@ -1305,6 +1313,44 @@ class SourceVisitor extends ThrowingAstVisitor {
   visitFunctionTypeAlias(FunctionTypeAlias node) {
     visitMetadata(node.metadata);
 
+    if (_formatter.fixes.contains(StyleFix.typedefs)) {
+      _simpleStatement(node, () {
+        // Inlined visitGenericTypeAlias
```

Comment:
All this copy/pasted code is a big concern for me. I try hard to minimize that because it tends to diverge over time and then similar code constructs end up formatting differently in different contexts.

Can you hoist this code out to private methods that are called in both places? If you scroll down past `visitYieldStatement()`, you'll see a lot of other methods that exist for this reason.

Here and below.

Reply:
Done.